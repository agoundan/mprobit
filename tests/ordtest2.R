options(error = expression(NULL))
library(mprobit)

# 4 repmeas for each cluster, 3 ordinal categories
dat=read.table("ordinalex.dat",header=T)
x=as.vector(dat[,2])
y=dat[,3]
ymat=matrix(y,100,4,T)
id=dat[,1]
x2=x*x

cat("\nordprobit.exch\n")
ord.exch <- ordprobit.exch(x,y,id,iprint=0)
print(ord.exch)
cat("============================================================\n")
cat("ordprobit.univar\n")
ord.univar<- ordprobit.univar(x,y,iprint=1)
print(ord.univar)
cat("============================================================\n")
# use of startpar
startp=c(ord.univar$cutpts,ord.univar$beta,0.5)
cat("ordprobit.exch with startpar\n")
print(startp)
ord.exch2 <- ordprobit.exch(x,y,id,iprint=0,startpar=startp)
print(ord.exch2)
cat("============================================================\n")

# compare ordinal logit or polr in library(MASS)
cat("ordlogit\n")
library(MASS)
ord.polr= polr(as.factor(y)~x)
print(ord.polr)
cat("============================================================\n")

# check 2 covariates
cat("\nordprobit.exch: x,x^2\n")
ord2.exch <- ordprobit.exch(cbind(x,x2),y,id,iprint=0)
print(ord2.exch)
cat("============================================================\n")
cat("ordprobit.univar: x,x^2\n")
ord2.univar<- ordprobit.univar(cbind(x,x2),y,iprint=1)
print(ord2.univar)
cat("============================================================\n")
cat("ordlogit: x,x^2\n")
ord2.polr= polr(as.factor(y)~x+x2)
print(ord2.polr)
cat("============================================================\n")

