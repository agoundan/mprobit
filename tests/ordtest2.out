
ordprobit.exch
$negloglik
[1] 724.186

$cutpts
[1] -0.4402640  0.3770676

$beta
[1] 0.4388846

$rho
[1] 0.6707258

$cov
              [,1]          [,2]          [,3]          [,4]
[1,]  0.0045090762  0.0032364272 -0.0003069651  0.0004705830
[2,]  0.0032364272  0.0045746241  0.0001821551 -0.0004877280
[3,] -0.0003069651  0.0001821551  0.0027797121 -0.0004684136
[4,]  0.0004705830 -0.0004877280 -0.0004684136  0.0013333861

============================================================
ordprobit.univar
iter= 1 , (with centered x's) cutp= -0.4303291 0.3920394 , b= 0.4278313 
         scorevec= -6.9722e-14 1.079137e-13 86.71818 

iter= 2 , (with centered x's) cutp= -0.4305042 0.3918036 , b= 0.432527 
         scorevec= 0.08882046 -0.2236458 0.9387407 

iter= 3 , (with centered x's) cutp= -0.4305042 0.3918035 , b= 0.4325278 
         scorevec= 5.096051e-05 -8.53638e-05 0.0001588693 

(with original x's) cutp= -0.4347107 0.387597 
nllk=  859.0873 
cutpts=  -0.4347107 0.387597 
beta=  0.4325278 
SEs :  0.0463587 0.04595953 0.0710764 

$negloglik
[1] 859.0873

$cutpts
[1] -0.4347107  0.3875970

$beta
[1] 0.4325278

$cov
              [,1]         [,2]          [,3]
[1,]  0.0021491292 0.0010851823 -0.0001802558
[2,]  0.0010851823 0.0021122786  0.0001006507
[3,] -0.0001802558 0.0001006507  0.0050518545

============================================================
ordprobit.exch with startpar
[1] -0.4347107  0.3875970  0.4325278  0.5000000
$negloglik
[1] 724.186

$cutpts
[1] -0.4402641  0.3770676

$beta
[1] 0.4388845

$rho
[1] 0.6707258

$cov
              [,1]          [,2]          [,3]          [,4]
[1,]  0.0048876781  0.0034754275 -0.0003453628  0.0004988589
[2,]  0.0034754275  0.0048573148  0.0003210357 -0.0005056633
[3,] -0.0003453628  0.0003210357  0.0032354905 -0.0004403820
[4,]  0.0004988589 -0.0005056633 -0.0004403820  0.0013832756

============================================================
ordlogit
Call:
polr(formula = as.factor(y) ~ x)

Coefficients:
       x 
-0.70184 

Intercepts:
       1|2        2|3 
-0.6999682  0.6319491 

Residual Deviance: 1718.494 
AIC: 1724.494 
============================================================

ordprobit.exch: x,x^2
$negloglik
[1] 724.1079

$cutpts
[1] -0.4275457  0.3898906

$beta
[1]  0.43866594 -0.04038954

$rho
[1] 0.6710153

$cov
              [,1]          [,2]          [,3]          [,4]          [,5]
[1,]  0.0061267932  0.0047445012 -4.038963e-04 -2.962933e-03  4.685886e-04
[2,]  0.0047445012  0.0060505042  3.245663e-04 -2.700853e-03 -4.803301e-04
[3,] -0.0004038963  0.0003245663  2.650456e-03  8.585342e-05 -4.212321e-04
[4,] -0.0029629331 -0.0027008526  8.585342e-05  8.837360e-03 -1.801011e-05
[5,]  0.0004685886 -0.0004803301 -4.212321e-04 -1.801011e-05  1.289627e-03

============================================================
ordprobit.univar: x,x^2
iter= 1 , (with centered x's) cutp= -0.4303289 0.3920396 , b= 0.4278276 -0.001035429 
         scorevec= -6.9722e-14 1.079137e-13 86.71818 -0.3704748 

iter= 2 , (with centered x's) cutp= -0.4305128 0.3917967 , b= 0.432522 -0.002017060 
         scorevec= 0.08720501 -0.2245867 0.9389597 -0.05400295 

iter= 3 , (with centered x's) cutp= -0.4305128 0.3917965 , b= 0.4325228 -0.002017605 
         scorevec= 3.227191e-05 -9.964143e-05 0.0001641558 -2.908385e-05 

(with original x's) cutp= -0.4340646 0.3882448 
nllk=  859.0872 
cutpts=  -0.4340646 0.3882448 
beta=  0.4325228 -0.002017605 
SEs :  0.06389903 0.06368822 0.07107737 0.1373201 

$negloglik
[1] 859.0872

$cutpts
                      
-0.4340646  0.3882448 

$beta
           x           x2 
 0.432522818 -0.002017605 

$cov
                                           x            x2
    0.0040830861  3.024105e-03 -1.951234e-04 -6.038886e-03
    0.0030241046  4.056189e-03  8.576443e-05 -6.054411e-03
x  -0.0001951234  8.576443e-05  5.051992e-03  4.603691e-05
x2 -0.0060388856 -6.054411e-03  4.603691e-05  1.885680e-02

============================================================
ordlogit: x,x^2
Call:
polr(formula = as.factor(y) ~ x + x2)

Coefficients:
          x          x2 
-0.70192161 -0.01805051 

Intercepts:
       1|2        2|3 
-0.7059579  0.6260678 

Residual Deviance: 1718.487 
AIC: 1726.487 
============================================================

